PageObjects/CartPage.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Playwright;
+using NUnit.Framework;
+using TechTalk.SpecFlow;
+
+namespace HW18_SpecFlow.PageObjects
+{
+    [Binding]
+    internal class CartPage
+    {
+        private readonly IPage page;
+
+        public CartPage(IPage page)
+        {
+            this.page = page;
+        }
+
+        public async Task VerifyPageUrl(string testPageUrl)
+        {
+            try
+            {
+                await page.WaitForURLAsync(testPageUrl);
+            }
+            catch (PlaywrightException e)
+            {
+                if (e.Message.Contains("crash"))
+                {
+                    Console.WriteLine("Page crashed: " + e.Message);
+                }
+            }
+        }
+
+        public async Task VerifyHeadingVisible(string heading)
+        {
+            var headingLocator = page.GetByRole(AriaRole.Heading, new() { Name = heading });
+            await Assertions.Expect(headingLocator).ToBeVisibleAsync();
+        }
+
+        public async Task VerifyProductAddedToCart(string addProductName)
```

Comment:
name should be just "productName". "Add" is already mentioned in the method name